test-network-function/networking/suite.go

```diff
@@ -188,23 +200,67 @@ func processContainerIpsPerNet(containerID *configsections.ContainerIdentifier,
 	netsUnderTest[netKey] = entry
 }
 
+func IsIPv4(address string) bool {
+	ip := net.ParseIP(address)
+	if ip == nil {
+		return false
+	}
+	if ip.To4() == nil {
+		return false
+	}
+	return true
+}
+
+func IsIPv6(address string) bool {
+	ip := net.ParseIP(address)
+	if ip == nil {
+		return false
+	}
+	if ip.To4() != nil {
+		return false
+	}
+	return true
+}
+
+func FilterIPListPerVersion(ipList []string, aIPVersion ipVersion) []string {
+	var filteredIPList []string
+	for _, aIP := range ipList {
+		if aIPVersion == IPv4 && IsIPv4(aIP) {
```

Comment:
No need to change this, but here's a personal opinion on this function: It has too many checks, I think. You could do the same with just one helper function `getIpVersion(ip string) ipVersion` :
```
for _, aIP := range ipList {
    if aIPVersion == getIpVersion(aIP) {
         filteredIpList = append(...)
    }
}


```
The caller should pass a valid IPs list (ipv4/ipv6). In case you really want to throw an error log like in line 233 for invalid ip strings, that can be done at the time where that ipList is built, not here.

Reply:
fixed